lib/ansible/module_utils/common/text/converters.py

```diff
@@ -251,7 +323,7 @@ def jsonify(data, **kwargs):
     return json.dumps(data, cls=_common_json._get_legacy_encoder(), _decode_bytes=True, **kwargs)
 
 
-def container_to_bytes(d, encoding='utf-8', errors='surrogate_or_strict'):
+def container_to_bytes(d: _t.Any, encoding: str = 'utf-8', errors: _ErrorHandlers = 'surrogate_or_strict') -> _t.Any:
```

Comment:
It looks like this is
```suggestion
def container_to_bytes(d: str | dict[bytes | str, bytes | str] | list[bytes | str] | tuple[bytes | str] | object, encoding: str = 'utf-8', errors: _ErrorHandlers = 'surrogate_or_strict') -> dict[bytes, bytes] | list[bytes] | tuple[bytes] | object:
```

Reply:
~~Why? It also accepts all other types (and simply passes them through). Maybe it should be `object` instead of `_t.Any`, but I don't think it should be anything else. (Unless we have some specific type for JSON values.)~~

I missed the `object`.

But then, still: why? These expressions can be both simplified to `object`. So you'd have `def container_to_bytes(d: object, ...) -> object:`.

Reply:
`object` isn't equvivalent to a union of more accurate types as those allow more behaviors than just `object`. I'm not entirely sure that it'd work. But you could try. Not sure about the return type, though. Returning `object` is very broad. The arg types should be broad but return types should be narrow.